src/libraries/System.Data.Common/src/System/Data/DataSet.cs

```diff
@@ -268,6 +268,8 @@ protected DataSet(SerializationInfo info, StreamingContext context, bool Constru
             DeserializeDataSet(info, context, remotingFormat, schemaSerializationMode);
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "Binary serialization is unsafe in general and is planned to be obsoleted. We do not want to mark interface or ctors of this class as unsafe as that would show many unnecessary warnings elsewhere.")]
```

Comment:
@vitek-karas @MichalStrehovsky @marek-safar - are you OK with these suppressions? According to https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#entirety-of-binaryformatter-type-obsolete-as-warning, these `GetObjectData` methods are scheduled for obsoletion in .NET 7, along with `ISerializable`. So going through the work of marking ISerializable.GetObjectData as `[RUC]` just doesn't seem worth it right now.

Reply:
cc @GrabYourPitchforks

Reply:
Spoke with Eric offline about this. Just to clarify: `ISerializable` is distinct from `BinaryFormatter`, though the two are commonly used together. Our primary focus is on removing `BinaryFormatter` from the ecosystem. That doc also talks about removing `ISerializable` from the ecosystem, but there's less immediate urgency around this, so the timelines might not match up 100%.

Regardless, I think suppressing this is the right move for now. 👍